### DISCUSSION TITLE
Contentment folder doesn't form itself when running on docking container

### DISCUSSION THREAD

**Discussion by eliranrefael** [leekelleher/umbraco-contentment#295] [Question] [answered]
Hey, thank you for your project, its very resourceful.
im using umbraco 10.3.2 with contentment 4.4 (not core), and for some unexplained reason, when i working with my project on a docker i cant see that contentment folder is formed under wwwroot/app_plugins and ofcourse i get ton of errors because of it.
when i run the project in other ways, like in IISExpress, the app_plugin is created and populated outside the wwwroot and everything working wonderfully.
i also have usync installed and it works fine on both envs...
any clues of why that could be happening?

thanks!

**Comment by leekelleher** [comment]
Hi @eliranrefael. As far as I'm aware, the `App_Plugins/Contentment` folder should be outside of the `wwwroot`.

I haven't used Docker, so I'm unsure what complexities that introduces.

With uSync, I do know that it uses the Razor Class Library approach to embed the web assets (e.g. the `App_Plugins` folder), which can later be deployed to inside the `wwwroot` folder, but Contentment doesn't follow the same pattern as that.

**Reply by eliranrefael** [reply]
thank you for your reply, would you recommend to include contentment folder in my source control and include it in my project?

**Reply by leekelleher** [reply]
From what I recall, when the web-app is compiled (whether locally or in a CI/CD build process) then the NuGet package should re-copy over the web asset files to the `App_Plugin/Contentment` target folder. Meaning that you shouldn't need them in your source control, but if you are experiencing issues, and adding them to your source control resolves that, then all good. 👍

**Reply by eliranrefael** [reply]
A little update,
its not releated to container or not, its a bug happening during the publish process, if the app_plugins does not exist from first place, it wont be copied to the publish folder, even if you enclude it in the csproj. further described here:
https://github.com/umbraco/Umbraco-CMS/issues/12572

i think maybe its related to when the folder is created in the process, if anyone has suggestions but adding the folder to the source control i will be glad to hear!

**Comment by leekelleher** [comment, marked as answer]
Closing this off, as I think this has been resolved since [Contentment v5.0](https://github.com/leekelleher/umbraco-contentment/releases/tag/5.0.0), with the supporting the using of a Razor Class Library.
